src/Agent.Worker/Build/ArtifactCommandExtension.cs

```diff
@@ -194,6 +211,62 @@ private void ProcessArtifactUploadCommand(IExecutionContext context, Dictionary<
                                                       propertyDictionary,
                                                       fullPath,
                                                       context.CancellationToken);
+            } else if (uploadService == ArtifactUploadSerivce.Drop)
+            {
+                commandContext.Task = UploadDropArtifactAsync(commandContext,
+                                                      WorkerUtilities.GetVssConnection(context),
+                                                      projectId,
+                                                      buildId.Value,
+                                                      artifactName,
+                                                      propertyDictionary,
+                                                      fullPath,
+                                                      context.CancellationToken);
+            }
+            else
+            {
+                throw new Exception(StringUtil.Loc("ArtifactUploadServiceNotSupported", uploadService));
+            }
+
+            context.AsyncCommands.Add(commandContext);
+        }
+
+        private void ProcessArtifactDownloadCommand(IExecutionContext context, Dictionary<string, string> eventProperties, string data)
+        {
+            ArgUtil.NotNull(context, nameof(context));
+            ArgUtil.NotNull(context.Endpoints, nameof(context.Endpoints));
+
+            Guid projectId = context.Variables.System_TeamProjectId ?? Guid.Empty;
+            ArgUtil.NotEmpty(projectId, nameof(projectId));
+
+            string localPath = data;
+            if (string.IsNullOrEmpty(localPath))
+            {
+                throw new Exception(StringUtil.Loc("ArtifactLocationRequired"));
+            }
+
```

Comment:
minor: For symmetry, I think it would be good to check `artifacttype` and `throw new Exception(...)` if it's not drop. You could add a comment that container download is implemented in typescript.